shell/platform/windows/window.h

```diff
@@ -249,6 +253,9 @@ class Window : public KeyboardManager::WindowDelegate {
   // Accessibility node that represents an alert.
   std::unique_ptr<ui::AXPlatformNodeWin> alert_node_;
 
+  // Guard against posthumous vtable access;
+  bool vtable_is_alive = true;
```

Comment:
This is spooky. Some potential alternative options:

1. Merge `FlutterWindow` and `Window` to make destruction simpler
2. Expose a `Destroy` method to destroy the window manually. This would need to be called before the `Window` is destructed to ensure no messages are receive post destruction.
3. Switch from an inheritance pattern to a delegate pattern so that the `Window` is destructed before the `FlutterWindow`

What are your thoughts?

Reply:
1. I do not see any reason why they need to be separate classes, though I am a little nervous about trying it. As long as it's safe, and no one will complain about needing both classes for some 3rd party or future change, this seems like a valid option.
2. We could also try this, though do we know from where we could call it, or also importantly, if we have any way to ensure it's called before destruction?
3. If `FlutterWindow` is our only use case for `Window`, merging the classes seems like a better idea.

Reply:
> 1. I do not see any reason why they need to be separate classes, though I am a little nervous about trying it. As long as it's safe, and no one will complain about needing both classes for some 3rd party or future change, this seems like a valid option.

This type hierarchy is a common win32 C++ pattern: https://learn.microsoft.com/windows/win32/learnwin32/managing-application-state-#an-object-oriented-approach

I'm not in love with this pattern though 😅

> 2. ... though do we know from where we could call it, or also importantly, if we have any way to ensure it's called before destruction?

One potential place might `FlutterDesktopViewControllerDestroy`. We could manually close the window before destroying our objects:

https://github.com/flutter/engine/blob/138a1ea9a69235342f479d61399fe7394fc1f3cb/shell/platform/windows/flutter_windows.cc#L90-L93

> 3. ... merging the classes seems like a better idea.

Agreed. Right now I'm leaning towards option 2, it seems like the least intrusive option.

/cc @cbracken For thoughts

Reply:
What do you think about moving the call to `Window::Destroy()` from `~Window` to `~FlutterWindow`? I just do not know how we can enforce within a test that window messages are received between the two destructors.

Reply:
In the interest of getting this landed, agreed that option 2 (make Destroy() protected, then move the destroy call down to the subclass) is probably the quickest fix. Once that's landed, let's follow up with a patch that implements option 1 (merge Window and FlutterWindow).

I don't think there's a strong case for keeping them separate. Now that the UWP embedder is gone let's use the [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) rule here and merge the two. If we ever find a good reason for a separation, we can look at that in the future.